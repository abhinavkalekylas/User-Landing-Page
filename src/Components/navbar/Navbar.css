nav {
    width: 100%;
}

.navbar {
    background: black;
    justify-content: space-between;
    margin-top: 0.25rem;
}

.navbar-brand {
    margin-left: 0.75rem;
    margin-right: 0.75rem;
}

form {
    display: flex;
    flex-direction: row;
}

.form-select {
    margin-top: 0.3em;
    margin-bottom: 0.3em;
}

.adduser > button {
    width: 120px;
    height: 45px;
    margin-left: 1rem;
    margin-right: 1rem;
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}