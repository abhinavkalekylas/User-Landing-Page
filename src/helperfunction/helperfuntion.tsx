export interface User {
  id: number;
  name: string;
  email: string;
  gender: GenderType;
  status: StatusType;
}

export enum GenderType {
  MALE = "male",
  FEMALE = "female",
}

export enum StatusType {
  ACTIVE = "active",
  INACTIVE = "inactive",
}
