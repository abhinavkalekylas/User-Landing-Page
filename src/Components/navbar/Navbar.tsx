import React, { useEffect, useState } from "react";
import { Button } from "react-bootstrap";
import "./Navbar.scss";
import Modaldata from "../modal/Modaldata";
import { GenderType, StatusType } from "../../modalfunction/Modal";
import { useDispatch } from "react-redux";
import { filterUser } from "../../redux/action/todo";

const Navbar = () => {
  // user state to show/hide modal
  const [addUserModal, setAddUserModal] = useState(false);

  const closeModal = () => setAddUserModal(false);
  const openModal = () => setAddUserModal(true);

  const dispatch = useDispatch();
  const [filter, setFilter] = useState({
    selectedGender: "",
    selectedStatus: "",
  });

  const handleChangeStatus = (e: {
    target: { name: string; value: string };
  }) => {
    setFilter((prevType) => ({
      ...prevType,
      selectedStatus: e.target.value,
    }));
  };

  const handleChangeGender = (e: {
    target: { name: string; value: string };
  }) => {
    setFilter((prevType) => ({
      ...prevType,
      selectedGender: e.target.value,
    }));
  };

  useEffect(() => {
    dispatch(filterUser(filter));
  }, [filter]);

  return (
    <nav className="navbar navbar-dark">
      <a href="/" className="navbar-brand ">
        User Landing Page
      </a>

      <form className="nav_form form-inline">
        <span className="filter_title">Filter by:</span>
        <select
          className="filter form-select form-select-lg gender_select"
          aria-label=".form-select-lg example"
          onChange={handleChangeGender}
        >
          <option value="">None</option>
          <option value={GenderType.MALE}>Male</option>
          <option value={GenderType.FEMALE}>Female</option>
        </select>
        <select
          className="filter form-select form-select-lg status_select"
          aria-label=".form-select-lg example"
          onChange={handleChangeStatus}
        >
          <option value="">None</option>
          <option value={StatusType.ACTIVE}>Active</option>
          <option value={StatusType.INACTIVE}>Inactive</option>
        </select>
        <div className="adduser">
          <Button
            color="success"
            className="btn_add btn btn-success btn-outline-light"
            onClick={openModal}
          >
            Add User
          </Button>
        </div>
      </form>
      <Modaldata addUserModal={addUserModal} closeModal={closeModal} />
    </nav>
  );
};

export default Navbar;
