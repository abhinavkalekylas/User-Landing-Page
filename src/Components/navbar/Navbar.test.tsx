import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { StatusType } from "../../helperfunction/helperfuntion";
import Navbar from "./Navbar";

let wrapper: any;
beforeEach(() => {
  wrapper = shallow(<Navbar />);
});

it("check nav element", () => {
  expect(wrapper.find("nav")).toBeTruthy();
});

// snapshot testing
it("renders correctly App component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

// test for select tag for filter
it("test filter select input", () => {
  wrapper
    .find(".filter")
    .simulate("change", { target: { value: StatusType.ACTIVE } });
  expect(wrapper.find(".filter").props().value).toBe(StatusType.ACTIVE);
});
