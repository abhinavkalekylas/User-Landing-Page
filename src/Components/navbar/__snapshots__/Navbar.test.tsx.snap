// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly App component 1`] = `
<nav
  className="navbar navbar-dark"
>
  <a
    className="navbar-brand "
    href="/"
  >
    User Landing Page
  </a>
  <form
    className="filter form-inline"
  >
    <select
      aria-label=".form-select-lg example"
      className="form-select form-select-lg"
    >
      <option
        value="none"
      >
        None
      </option>
      <option
        value="active"
      >
        Active
      </option>
      <option
        value="inactive"
      >
        Inactive
      </option>
      <option
        value="male"
      >
        Male
      </option>
      <option
        value="female"
      >
        Female
      </option>
    </select>
    <div
      className="adduser"
    >
      <Button
        active={false}
        className="btn_add btn btn-success btn-outline-light"
        color="success"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Add User
      </Button>
    </div>
  </form>
  <Modaldata
    handleClose={[Function]}
    showModal={false}
  />
</nav>
`;
